### PR TITLE
Automatic header detection tests

### PR DESCRIPTION
**Pull request by muhanadz**
Added test for the new automatic column type (header) detection behavior from https://github.com/ome/omero-metadata/pull/67. The test will create an example `CSV` inclusive of the different combinations a user can use in a temp location which is passed into the new `detect_headers` function. The output is then checked with the expected output of said columns.

This would be a good place to start when we want to change the behavior of the automatic detection (eg column plate_id should be detected as type `plate` instead of the current `l`)

### CODE REVIEW

File: test/integration/detection/test_automatic_header.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        'project': [1, 2, 3],
+        'dataset': [1, 2, 3],
+        'plate': [1, 2, 3],
+        'well': [1, 2, 3],
+        'image': [1, 2, 3],
+        'roi': [1, 2, 3],
+        'measurement 1': [11, 22, 33],
+        'measurement 2': [0.1, 0.2, 0.3],
+        'measurement 3': ['a', 'b', 'c'],
+        'measurement 4': [True, True, False]
```
Comment (sbesson) [inline test/integration/detection/test_automatic_header.py:42]: Nice coverage of the primary column types. I would add at least one extra case with mixed values .

**Review by sbesson — CHANGES_REQUESTED** [review]
Looks good. A few suggestions of extra tests to cover all the scenarios introduced in #67.

I would also propose to migrate the file under `test/unit/test_automatic_header.py` as it does not require a working OMERO.server/omero-test-infra infrastructure.

File: test/integration/detection/test_automatic_header.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from omero_metadata.cli import MetadataControl
+import pandas as pd
+import tempfile
+
+
+def test_detect_headers():
+    d = {
+        'project_name': ['a', 'b', 'c'],
+        'dataset_name': ['a', 'b', 'c'],
+        'plate_name': ['a', 'b', 'c'],
```
[the remark this answers is not in the document]
Comment (muhanadz) [reply test/integration/detection/test_automatic_header.py]: Good idea, added and addressed in [654195b](https://github.com/ome/omero-metadata/pull/72/commits/654195b27aa87a336cb4d79d6462a7e2a5fd05fc)

File: test/integration/detection/test_automatic_header.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        'well_name': ['a', 'b', 'c'],
+        'image_name': ['a', 'b', 'c'],
+        'roi_name': ['a', 'b', 'c'],
+        'project_id': [1, 2, 3],
+        'dataset_id': [1, 2, 3],
+        'plate_id': [1, 2, 3],
+        'well_id': [1, 2, 3],
+        'image_id': [1, 2, 3],
+        'roi_id': [1, 2, 3],
```
[the remark this answers is not in the document]
Comment (muhanadz) [reply test/integration/detection/test_automatic_header.py]: Addressed in [654195b](https://github.com/ome/omero-metadata/pull/72/commits/654195b27aa87a336cb4d79d6462a7e2a5fd05fc)

File: test/integration/detection/test_automatic_header.py (hunk 1 above)
Comment (muhanadz) [reply test/integration/detection/test_automatic_header.py]: Addressed in [654195b](https://github.com/ome/omero-metadata/pull/72/commits/654195b27aa87a336cb4d79d6462a7e2a5fd05fc)

**Review by sbesson — APPROVED** [review]

**Pull request merged**